### PR TITLE
:bug: Schedule editor: Linked transactions table collapses

### PR DESCRIPTION
**Pull request by trevdor**
### Problem
h/t @Kidglove57 for [noticing this](https://github.com/actualbudget/actual/pull/1501#issuecomment-1686770556). 
The Schedules table in the Link Schedules modal gets squished to 0px height by the AutoSizer in `<Table>`, even if it has rows in it. In fact, the empty state "No transactions found" is the only situation where this table renders with a proper height.

Screenshot 2023-08-22 at 10 10 33 PM

### Solution
1. Fix the empty state to be able to render whether we're looking at Linked or Matched transactions.
2. Give `<Table>` a flex-basis equal to the row height multiplied by the number of rows, but let it flex grow AND shrink.

### Concerns
I don't like messing with `<Table>`, since it's used for Accounts transactions lists and several other places. But I verified the height comes out as expected in every spot where it's used. Large transaction lists may get a flex basis of 6000px+, but they flex shrink to fit within the virtualized list container.

Definitely open to alternate suggestions here.

### CODE REVIEW

File: packages/desktop-client/src/components/table.tsx
Comment (trevdor) [inline packages/desktop-client/src/components/table.tsx:1116]: Primary change to avoid a zero-height table.

File: packages/desktop-client/src/components/schedules/EditSchedule.js
Comment (trevdor) [inline packages/desktop-client/src/components/schedules/EditSchedule.js:737]: Here's the bit for displaying an message when "Linked transactions" is selected.

**Comment by MatissJanis** [comment]
@Kidglove57 is the demo link fixing the issue you observed?

**Review by MatissJanis — APPROVED** [review]
Code looks good; also spotted no issues when doing manual tests.

Lets just wait on @Kidglove57 confirmation before merging.

**Comment by Kidglove57** [comment]
@MatissJanis I have tried with a copy of my production file and everything is displaying exactly as I would expect (and scrolls for a very long list). Thank you both!

**Pull request merged**
